### PR TITLE
Fixes for missing events in calendar tag, refactoring, update test cases.

### PR DESCRIPTION
**Pull request by cyberstar**
* Fixes event manager ``.ongoing()`` with the correct lookup params
* Fixes events which were started before first date of calendar tag or end date was null
* Refactoring of calendar tag
* Update for test cases
* Minor refactoring of test cases

### CODE REVIEW

File: aldryn_events/utils.py [hunk 1]
```diff
@@ -185,31 +205,35 @@ def build_calendar(year, month, language, namespace=None):
 
     # Get a list of all dates in this month (with pre/succeeding for nice
     # layout)
-    monthdates = [(x, []) for x in get_monthdates(month, year)]
+    monthdates = [x for x in get_monthdates(month, year)]
```
Comment (cyberstar) [inline aldryn_events/utils.py]: maybe list() would be more preferable.

File: aldryn_events/utils.py [hunk 2]
```diff
@@ -61,7 +61,7 @@ def build_events_by_year(events, **config):
         if year not in events_by_year:
             events_by_year[year] = {
                 'year': year,
-                'date': date(year, 1, 1),
+                'date': datetime.date(year, 1, 1),
```
Comment (czpython) [inline aldryn_events/utils.py]: notice that a lot of calls to datetime.date() can accumulate, best to do something like ``date = datetime.date`` in the local scope of the function.

File: aldryn_events/utils.py [hunk 3]
```diff
@@ -175,6 +175,50 @@ def get_monthdates(month, year):
...
+def update_monthdates(monthdates, event, first_date, last_date):
+    """
+    Updates monthdates events in place. Returns monthdates.
+    :param monthdates: OrderedDict to update
+    :param event: Event that should be processed
+    :param first_date: first date of monthdates
+    :param last_date: last date of monthdates
+    :return: OrderedDict, link to monthdates (updated in place!)
+    """
+    if event.start_date < first_date:
```
Comment (czpython) [inline aldryn_events/utils.py]: python ftw ``day = max(event.start_date, first_date)``

**Comment by mkoistinen** [comment]
LGTM

**Pull request merged**